src/esrs/aggregate.rs

```diff
@@ -80,72 +70,57 @@ pub trait AggregateManager: Aggregate {
     /// Validates and handles the command onto the given state, and then passes the events to the store.
     async fn handle_command(
         &self,
-        aggregate_state: AggregateState<Self::State>,
+        mut aggregate_state: AggregateState<Self::State>,
         command: Self::Command,
-    ) -> Result<AggregateState<Self::State>, Self::Error> {
+    ) -> Result<(), Self::Error> {
```

Comment:
If you passed in a **_locked_** aggregate_state, we can only guarantee nothing else is mutating it in the store until we **_call_** `store_events`: by the time that Future resolves and we get back the vec of stored events, another process might have already handled another command and therefore added extra events on top of what the current process has in memory!

To make this race condition more explicit, we just return empty here. Downstream, after you handle your command, you have to explicitly reload the state yourself, which guarantees you get the latest state from the store (which might not be exactly what you just persisted!)